### DISCUSSION TITLE
Does ksmbd support file names with special characters

### DISCUSSION THREAD

**Discussion by rickysarraf** [cifsd-team/ksmbd-tools#320] [General]
Does ksmbd support file names with special characters ?

Special characters documented as in: https://web.archive.org/web/20150619005746/https://support.microsoft.com/en-us/kb/177506

I have lots of files with names having the `:` character. It works all well with `smbd`.

But with `ksmbd`,  when doing an `ls -lh` on them, I get the `No such file or directory`. The other files, that do not have the special `:` character, list all proper.

This is seen with:

Linux: 6.11
Distro: Debian Unstable
ksmbd-utils: 3.5.2-3

## Steps to reproduce

* On ksmbd server, create files with `:` in their names
* Mount on ksmbd client
* `ls -lh` on the ksmbd share

**Comment by namjaejeon** [comment]
It should be worked.

Can you tell me your filenames ?
because it show correctly when ls -lh on my setup.

root@linkinjeon-System-Product-Name:~# ls -lh /mnt/test/
total 0
-rwxr-xr-x 1 root root 0  9월  6 17:40 :a
-rwxr-xr-x 1 root root 0  9월  6 17:40 a:
-rwxr-xr-x 1 root root 0  9월  6 17:40 a:b
